.eslintrc.cjs

```diff
@@ -45,6 +45,18 @@ const config = {
       { prefer: 'type-imports' },
     ],
     '@typescript-eslint/explicit-module-boundary-types': 'off',
+    '@typescript-eslint/naming-convention': [
+      'warn',
+      {
+        selector: 'typeParameter',
+        format: ['PascalCase'],
+        leadingUnderscore: 'allow',
+        custom: {
+          regex: '^(T|\\$)[A-Z][a-zA-Z]+[0-9]*$',
+          match: true,
```

Comment:
hmm why would we allow leading underscores on a type param? I've set this in other projects to:

```
{
    selector: 'typeParameter',
    format: ['PascalCase'],
    leadingUnderscore: 'forbid',
    trailingUnderscore: 'forbid',
    custom: {
        regex: '^(T|\\$)([A-Z][a-z]+)*$',
        match: true,
    },
},
```

Reply:
> hmm why would we allow leading underscores on a type param?

I banned underscores, and simplified regex in [77af7b3](https://github.com/TanStack/query/pull/6684/commits/77af7b3dde36f11583e35566a4b59ebbae51f43a)